.github/workflows/linux.yml

```diff
@@ -422,8 +422,10 @@ jobs:
       - name: Setup system dependencies
         run: |
           sudo apt-get -y update
-          # `python3-dev` yields Python 3.10 on Ubuntu 22.04
-          sudo apt install -y python3-dev python3.11-dev ninja-build pkg-config libatlas-base-dev liblapack-dev
+          sudo apt install software-properties-common
+          sudo add-apt-repository ppa:deadsnakes/ppa
```

Comment:
This isn't desirable, `deadsnakes` isn't as stable as the main Ubuntu package repositories.

It should be fine to keep using `python3-dev`, that installs 3.12 in Ubuntu 24.04. The job doesn't need to run on the lowest supported Python version, the only point is to check a distro-supplied Python interpreter.

Reply:
I'll check locally before pushing an update

Reply:
Ah I see the problem, it doesn't have to be the lowest supported version, but it should be a non-default version. And Ubuntu 24.04 doesn't seem to have any. `noble-updates` may gain `3.13` at some point, but until then maybe this is the best option (at least easier than switching to a different distro).